### ISSUE TITLE
Reformat ternary operator

### ISSUE THREAD

**Issue by mjbvz** [microsoft/TypeScript#14494]
_From @lijunray on March 6, 2017 22:4_

- VSCode Version: 1.9.1
- OS Version: MacOS 10.12.3

Steps to Reproduce:

The default reformatting for ternary operator is working unexpectedly.

What I want:

[![enter image description here][1]][1]

After Reformatting:

[![enter image description here][2]][2]

How can I config the reformatting so that it works as I expect?

  [1]: https://i.stack.imgur.com/vnwGM.png
  [2]: https://i.stack.imgur.com/yhZei.png

_Copied from original issue: Microsoft/vscode#22110_

**Comment by mjbvz** [comment]
Example code:

```js
const test = (a) => (
	a === '1' ? (
		10 
	) : (
		12
	)
)
```

Format request:

```
[Trace - 2:09:16 PM] Sending request: format (187). Response expected: yes. Current queue length: 0
Arguments: {
    "file": "untitled:Untitled-6",
    "line": 1,
    "offset": 1,
    "endLine": 7,
    "endOffset": 2
}
[Trace - 2:09:16 PM] Response received: format (187). Request took 3 ms. Success: true 
Result: [
    {
        "start": {
            "line": 3,
            "offset": 5
        },
        "end": {
            "line": 3,
            "offset": 6
        },
        "newText": ""
    },
    {
        "start": {
            "line": 5,
            "offset": 1
        },
        "end": {
            "line": 5,
            "offset": 3
        },
        "newText": "\t\t\t"
    },
    {
        "start": {
            "line": 6,
            "offset": 1
        },
        "end": {
            "line": 6,
            "offset": 2
        },
        "newText": "\t\t"
    }
]
```

**Comment by jamesst20** [comment]
This bug is still present as of 1.17.2 (latest version)

**Comment by hipstersmoothie** [comment]
I def still see this bug

**Comment by reujab** [comment]
I've discovered a slightly less ugly workaround by using two parentheses:

```javascript
const test = (a) => (
	a === '1' ? (
		10 
	) : ((
		12
	))
)
```

I hope to see this fixed soon.

**Comment by ranisalt** [comment]
Still broken

**Comment by marnixhoh** [comment]
Still broken... Is there any progress on this?

**Comment by marnixhoh** [comment]
Still broken... I assume given how old this issue is that it will never get fixed? :(

**Comment by jamesst20** [comment]
Yeah.. I don't get how Microsoft think. TypeScript is their own technogy and so is VSCode.. it's been 3 years soon.

**Comment by isaacalves** [comment]
What is everyone doing in regards to this?

- disable `formatOnSave` entirely for TypeScript files?
- refactor ternary operators and use something else?
- live with the pain of having to deal with unwanted automatic changes in formatting (for example, committing these changes by mistake and then having to fix them)?

**Comment by ranisalt** [comment]
@isaacalves prettier

**Comment by jamesst20** [comment]
@isaacalves I use WebStorm instead

**Comment by isaacalves** [comment]
I've just found out that if you have the `?` of the ternary operator in a new line (instead of in the same line of the conditional), the formatting will be like this:

```
const test = (a) => (
    a === '1'
        ? (
            10
        ) : (
            12
        )
)
```

That seems to be ideal actually, and probably related to [this](**url**)
By the way I use prettier, and VS Code.

**Comment by RyanCavanaugh** [comment]
Closing language service bugs related to the 6.0 implementation. For more information, see #62827
